### PR TITLE
Added support for Mongoid 5, 6 and 7 relations

### PR DESCRIPTION
**Pull request by dblock**
Actually closes https://github.com/rmosolgo/graphql-ruby/issues/938, enabling plain `has_many` and similar relations to be used as relay connection types. These are no longer `Mongoid::Criteria` since Mongoid 5.

### CODE REVIEW

**Comment by dblock** [comment]
💚 and ready to merge @rmosolgo

File: gemfiles/rails_5.1.gemfile [hunk 1]
```diff
@@ -15,9 +15,4 @@ gem "puma"
 gem "capybara", "~> 2.13"
 gem "selenium-webdriver"
 
-group :jekyll_plugins do
-  gem "jekyll-algolia"
-  gem "jekyll-redirect-from"
-end
-
```
Comment (dblock) [inline gemfiles/rails_5.1.gemfile]: Removed these blocks as in other gemfiles. The `jekyll-algolia` gem requires ruby 2.3.0, and `graphql-ruby` still tests with 2.2.8, so it was producing errors from the auto-generated appraisals. I suspect somone had removed these from the earlier rails version by hand, and they aren't necessary for the tests anyway.

File: spec/graphql/relay/mongo_relation_connection_spec.rb [hunk 2]
```diff
@@ -482,4 +491,49 @@ def get_names(result, field_name)
     connection = GraphQL::Relay::BaseConnection.connection_for_nodes(relation)
     assert_equal GraphQL::Relay::MongoRelationConnection, connection
   end
+
+  describe "relations" do
+    let(:query_string) {%|
```
Comment (dblock) [inline spec/graphql/relay/mongo_relation_connection_spec.rb]: This one is quick and dirty. Happy to iterate on a smaller query if you care, but could use some help in writing one.

**Comment by rmosolgo** [comment]
🎊  This looks awesome, thanks for your work here!

**Pull request merged**